src/components/model-authoring/edit-name.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from "react";
+import { useState } from "react";
+
+import * as css from "./edit-name.scss";
+
+interface IProps {
+  name: string;
+  saveName: (name: string) => void;
+}
+
+export const EditName = ({ name, saveName }: IProps) => {
+  const [newName, setNewName] = useState<string>(name);
+  const [disabled, setDisabled] = useState<boolean>(true);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewName(e.target.value);
+  }
+
+  const handleClick = () => {
+    setDisabled(false);
+  }
+
+  const handleSave = () => {
+    saveName(newName);
+    setDisabled(true);
+  }
+
+  return (
+    <>
+      <input defaultValue={name} disabled={disabled} onChange={handleChange} />
```

Comment:
You are probably getting React warnings in your console because you have both a `defaultValue` and `onChange` handler.  You should switch to `value={newName}` and that should fix it.